### PR TITLE
fix compiler error of address will never be NULL

### PR DESCRIPTION
**Pull request by xiacunshun**
### Description 
_Add a comprehensive description of proposed changes_

Fixes # - _issue number(s) if exists_

- [ ] - git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [1 ] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [1 ] added - _required for new features and some bug fixes_
- [ ] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [1 ] not needed

### Breaks backward compatibility
- [ ] Yes
- [ 1] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

**Comment by xiacunshun** [comment]
fix https://github.com/oneapi-src/oneTBB/issues/1180

File: test/common/doctest.h [hunk 1]
```diff
@@ -1555,10 +1555,12 @@ DOCTEST_CLANG_SUPPRESS_WARNING_WITH_PUSH("-Wunused-comparison")
                 , m_at(at) {}
 
         DOCTEST_NOINLINE operator Result() {
+DOCTEST_GCC_SUPPRESS_WARNING_WITH_PUSH("-Waddress")
```
Comment (pavelkumbrasev) [inline test/common/doctest.h:1558]:
```suggestion
// TODO: upstream the change to doctest : Work-around for the warning: 'address will never be NULL'
DOCTEST_GCC_SUPPRESS_WARNING_WITH_PUSH("-Waddress")
```

Comment (xiacunshun) [reply test/common/doctest.h:1558]: done

File: test/common/doctest.h [hunk 2]
```diff
@@ -1555,10 +1555,13 @@ DOCTEST_CLANG_SUPPRESS_WARNING_WITH_PUSH("-Wunused-comparison")
...
+// TODO: upstream the change to doctest : Work-around for the warning: 'address will never be NULL'
```
Comment (pavelkumbrasev) [inline test/common/doctest.h:1559]:
```suggestion
// TODO: upstream the change to doctest : Work-around for the warning: 'address will never be NULL'
DOCTEST_GCC_SUPPRESS_WARNING_WITH_PUSH("-Waddress")
```
Should be this ordering

Comment (xiacunshun) [reply test/common/doctest.h]: Ok. Fixed it.

**Comment by xiacunshun** [comment]
I saw a CI test failed. Does it matter?

**Comment by dnmokhov** [comment]
> I saw a CI test failed. Does it matter?

No, it is was an unrelated CI issue. I have relaunched the test.

**Comment by pavelkumbrasev** [comment]
> I saw a CI test failed. Does it matter?

If you're mentioning `documentation build` it's probably some internal problem unrelated to the fix.

**Review by pavelkumbrasev — APPROVED** [review]

**Pull request merged**